Problem-2.py

```diff
@@ -0,0 +1,41 @@
+#  15. 3Sum
+'''
+Leetcode all test cases passed: Yes
+class Solution:
+    def threeSum(self, nums: List[int]) -> List[List[int]]:
+        n is the length of nums
+        Space Complexity: O(1)
```

Comment:
SC depends on the implementation of sorting algorithm. It can be anywhere from O(log n) to O(n)